### ISSUE TITLE
Follow up on def vs val in #940

### ISSUE THREAD

**Issue by ceedubs** [typelevel/cats#965]
In #940 some implicit instances that seemed like they should be able to be `val`s were forced to be `def`s. It would be nice to follow up on that - to try to figure out why that was and whether we can do something about it.

cc @djspiewak

**Comment by djspiewak** [comment]
Here's a quick instance where this happens: https://github.com/typelevel/cats/blob/7cdd66136a0ddd13522a3f6492627ec38d657dc0/core/src/main/scala/cats/data/OptionT.scala#L140  Haven't tried in a while, but when I first wrote this line, swapping it from a `def` to a `val` made the implicit it provides unresolvable.

**Comment by ceedubs** [comment]
@djspiewak yes, I saw the same behavior when I didn't believe you and tried to change it to a `val` :P

**Comment by djspiewak** [comment]
@ceedubs Doubter…  ;-)

**Comment by ceedubs** [comment]
This is no longer relevant after #1621.
